src/ILCompiler.RyuJit/src/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -484,5 +505,220 @@ private InfoAccessType constructStringLiteral(CORINFO_MODULE_STRUCT_* module, md
             ppValue = (void*)ObjectToHandle(stringObject);
             return stringObject.RepresentsIndirectionCell ? InfoAccessType.IAT_PVALUE : InfoAccessType.IAT_VALUE;
         }
+
+        private void setVars(CORINFO_METHOD_STRUCT_* ftn, uint cVars, NativeVarInfo* vars)
+        {
+            if (_localSlotToInfoMap == null && _parameterIndexToNameMap == null)
+            {
+                return;
+            }
+
+            uint paramCount = (_parameterIndexToNameMap == null) ? 0 : (uint)_parameterIndexToNameMap.Count;
+            Dictionary<uint, DebugVarInfo> debugVars = new Dictionary<uint, DebugVarInfo>();
+
+            for (int i = 0; i < cVars; i++)
+            {
+                NativeVarInfo nativeVarInfo = vars[i];
+
+                if (nativeVarInfo.varNumber < paramCount)
+                {
+                    string name = _parameterIndexToNameMap[nativeVarInfo.varNumber];
+                    updateDebugVarInfo(debugVars, name, true, nativeVarInfo);
+                }
+                else if (_localSlotToInfoMap != null)
```

Comment:
What should happen if _localSlotToInfoMap is null here? Do we just drop/skip processing vars[i]?

Reply:
`SetLocalVariables` sets it below in this file. `_localSlotToInfoMap` would be null if the method has no locals. Jit compilation figures out all the locals / parameter types ahead of actual method compilation.